samples/Identity.ExternalClaims/Startup.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Identity.ExternalClaims.Data;
+using Identity.ExternalClaims.Services;
+using System.Security.Claims;
+
+namespace Identity.ExternalClaims
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<ApplicationDbContext>(options =>
+                options.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=aspnet-Identity.ExternalClaims-53bc9b9d-9d6a-45d4-8429-2a2761773502;Trusted_Connection=True;MultipleActiveResultSets=true"));
+
+            services.AddIdentity<ApplicationUser, IdentityRole>()
+                .AddEntityFrameworkStores<ApplicationDbContext>()
+                .AddDefaultTokenProviders();
+
+            services.AddAuthentication().AddGoogle(o =>
+            {
+                // Configure your auth
+                o.ClientId = "<yourid>";
+                o.ClientSecret = "<yoursecret>";
+                o.Scope.Add("https://www.googleapis.com/auth/plus.me");
+                o.ClaimActions.MapJsonKey(ClaimTypes.Gender, "gender");
+                o.Events.OnCreatingTicket = ctx =>
+                {
+                    ctx.Properties.StoreTokens(new AuthenticationToken[] { new AuthenticationToken() { Name = "AccessToken", Value = ctx.AccessToken } });
```

Comment:
I'm still not sure why you're doing this manually rather than using SaveTokens.
https://github.com/aspnet/Security/blob/13e2a16b319b94918e626659ead4322737f58a8d/src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationOptions.cs#L117

Reply:
Well originally this sample was trying to focus on how to get things from the external ticket into the identity cookie.  I wasn't intending to show much of the various sugar/tokens stuff since that papers over how things actually work, but they make sense in this sample as well, since they do tackle many of the common tasks.  I can demonstrate save tokens for the access tokens, and just add a custom claim instead to demonstrate the creating ticket